lnwire/onion_error.go

```diff
@@ -835,9 +843,33 @@ func (f *FailFeeInsufficient) Decode(r io.Reader, pver uint32) error {
 	}
 
 	f.Update = ChannelUpdate{}
-	return parseChannelUpdateCompatabilityMode(
+	err := parseChannelUpdateCompatabilityMode(
 		r, length, &f.Update, pver,
 	)
+	if err != nil {
+		return err
+	}
+
+	// Decode tlv extension and set incoming update if present and valid.
+	var incomingUpdate ChannelUpdate
+	incomingUpdateRecord := incomingUpdate.Record()
+
+	tlvStream, err := tlv.NewStream(incomingUpdateRecord)
+	if err != nil {
+		return err
+	}
+
+	types, err := tlvStream.DecodeWithParsedTypes(r)
```

Comment:
This should be using `DecodeWithParsedTypesP2P` to avoid a fake record with a maliciously long length